feature-libs/asm/components/customer-emulation/customer-emulation.component.spec.ts

```diff
@@ -74,11 +104,67 @@ describe('CustomerEmulationComponent', () => {
     fixture.detectChanges();
 
     //Click button
-    const endSessionButton = fixture.debugElement.query(By.css('button'));
+    const endSessionButton = fixture.debugElement.query(
+      By.css('button[formControlName="logoutCustomer"]')
+    );
     spyOn(asmComponentService, 'logoutCustomer').and.stub();
     endSessionButton.nativeElement.click();
 
     //assert
     expect(asmComponentService.logoutCustomer).toHaveBeenCalled();
   });
+
+  it('should assign cart to customer', () => {
+    const testUser = { uid: 'user@test.com', name: 'Test User' } as User;
+    const prevActiveCartId = '00001122';
+    const testCartId = '00001234';
+    const assignCartToCustomerButton = fixture.debugElement.query(
+      By.css('button.assignCartToCustomer')
+    );
+    spyOn(userService, 'get').and.returnValue(of(testUser));
+
+    spyOn(occAsmAdapter, 'bindCart').and.returnValue(of());
+    spyOn(activeCartFacade, 'getActiveCartId').and.returnValue(
+      of(prevActiveCartId)
+    );
+
+    component.ngOnInit();
+    fixture.detectChanges();
+
+    // check that cart id entered matches
+    expect(component.cartId.value).toEqual(prevActiveCartId);
+
+    // check that cart id exists
+    expect(component.cartIdExists).toBeTruthy();
+
+    // check that assign to cart button is enabled
+    expect(
+      assignCartToCustomerButton.nativeElement.getAttribute('disabled')
+    ).toBeFalsy();
+
+    // clear entered cart id
+    component.cartId.setValue('');
+
+    // check that cart id DOES NOT exists
+    expect(component.cartIdExists).toBeFalsy();
+
+    // check that assign to cart button is disabled
+    expect(
+      assignCartToCustomerButton.nativeElement.getAttribute('disabled')
+    ).toBeNull();
+
+    // set cart number to assign
+    component.cartId.setValue(testCartId);
+
+    // check that cart id exists
+    expect(component.cartIdExists).toBeTruthy();
+
+    // check that cart id entered matches
+    expect(component.cartId.value).toEqual(testCartId);
+
+    // check that assign to cart button is enabled
+    expect(
+      assignCartToCustomerButton.nativeElement.getAttribute('disabled')
+    ).toBeFalsy();
+  });
```

Comment:
At fist glance, the unit test seems to be written like an e2e test.  Can the design be adjusted to test individual behaviours? (like the component doesn't submit the form if the field is empty.)